apps/docs-snippets/src/guide/transactions/transaction-parameters.test.ts

```diff
@@ -0,0 +1,85 @@
+import type { Contract, Provider, TxParams } from 'fuels';
+import { BN, PolicyType, ScriptTransactionRequest, bn } from 'fuels';
+import { expectTypeOf } from 'vitest';
+
+import {
+  DocSnippetProjectsEnum,
+  getDocsSnippetsForcProject,
+} from '../../../test/fixtures/forc-projects';
+import { createAndDeployContractFromProject } from '../../utils';
+
+/**
+ * @group node
+ */
+describe(__filename, () => {
+  let contract: Contract;
+  let provider: Provider;
+  let gasPrice: BN;
+
+  const { binHexlified: scriptBytecode } = getDocsSnippetsForcProject(
+    DocSnippetProjectsEnum.SUM_SCRIPT
+  );
+
+  beforeAll(async () => {
+    contract = await createAndDeployContractFromProject(DocSnippetProjectsEnum.COUNTER);
+    provider = contract.provider;
+    gasPrice = provider.getGasConfig().minGasPrice;
+  });
+
+  it('matches tx param types', () => {
+    // #region transaction-parameters-1
+    // import { TxParams, bn } from 'fuels';
+
+    const txParams: TxParams = {
+      gasPrice: bn(1), // BigNumberish
+      gasLimit: bn(1), // BigNumberish
+      maturity: 1, // number or undefined
+      maxFee: bn(1), // BigNumberish or undefined
+      witnessLimit: bn(1), // BigNumberish or undefined
+      variableOutputs: 1, // number
+    };
+    // #endregion transaction-parameters-1
```

Comment:
I think explaining each parameter would be beneficial like [here](https://docs.fuel.network/docs/fuels-ts/cli/config-file/).

You've outlined 3 parameters in the [list](https://github.com/FuelLabs/fuels-ts/pull/1961/files#diff-58920915a7f022eb88089c997f230b4268730b90c9fa2f848e9d130bfe0b249fR5), but there are 6 in total? What do the other ones do?

Reply:
https://github.com/FuelLabs/fuels-ts/pull/1961/commits/14b3ca3b74bae11ec1850ae55c3fc29b5d38b641